packages/nuxt/src/pages/utils.ts

```diff
@@ -63,6 +64,12 @@ export function generateRoutesFromFiles (files: string[], pagesDir: string): Nux
       children: []
     }
 
+    const fileContent = fs.readFileSync(resolve(pagesDir, file), 'utf-8')
```

Comment:
We should also support reading from virtual fs (accessible from `nuxt.vfs`).

Reply:
Not 100% sure I got it right but this has been added as well